### PR TITLE
Substitute argparse for getopt in regex_generator.py

### PR DESCRIPTION
**Pull request by msk**

### CODE REVIEW

**Comment by msk** [comment]
[argparse in sniffles](https://trello.com/c/rR2IwvGO/11-argparse-in-sniffles)

**Comment by msk** [comment]
[argparse in sniffles](https://trello.com/c/rR2IwvGO/11-argparse-in-sniffles)

File: sniffles/regex_generator.py [hunk 1]
```diff
@@ -25,78 +25,140 @@
...
+    is the probability that that slot will be chosen.  For example,
+    the base RE structural type distribution has three slots.  The
+    first slot is the probability that the next structure type is
+    a character (where a character can be a letter, digit, binary, ASCII,
+    or substitution class (like \\w).  The second slot is for character
+    classes like [ab@%], [^123], or [a-z].  The final slot is the probability
+    of alternation occuring like (ab|cd).  With these three slots you can tune
+    how often you would like the structures to appear in your regular
+    expressions.  For example, python3.4 -c 10 -t "80,10,10" would create
+    10 regular expressions where 80% of the structures used would be
```
Comment (vvalgenti) [inline sniffles/regex_generator.py]: I noticed below we have %% for percent, but here only one.  Is that significant?

File: sniffles/regex_generator.py [hunk 2]
```diff
@@ -453,92 +515,8 @@ def usage():
     [-o options chance] [-R repetition chance] [-r repetition distribution]
     [-t re structural type distribution] [-?] [-g]
 
-    -C \t Character Distribution: This sets the possibility of seeing
```
Comment (vvalgenti) [inline sniffles/regex_generator.py]: Do we still need the usage statement?

File: sniffles/regex_generator.py (hunk 1 above)
Comment (msk) [reply sniffles/regex_generator.py]: Yes. For some reason, `description` is treated as a normal string while `help` is treated as a format string, where `%` has a special meaning.

**Pull request merged**